sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/text/DefaultSource.scala

```diff
@@ -92,17 +100,25 @@ private[sql] class TextRelation(
     sqlContext.sparkContext.hadoopRDD(
       conf.asInstanceOf[JobConf], classOf[TextInputFormat], classOf[LongWritable], classOf[Text])
       .mapPartitions { iter =>
+        val bufferHolder = new BufferHolder
+        val unsafeRowWriter = new UnsafeRowWriter
+        val unsafeRow = new UnsafeRow
+
         var buffer = new Array[Byte](1024)
-        val row = new GenericMutableRow(1)
         iter.map { case (_, line) =>
           if (line.getLength > buffer.length) {
             buffer = new Array[Byte](line.getLength)
           }
           System.arraycopy(line.getBytes, 0, buffer, 0, line.getLength)
```

Comment:
Do we still need this copy here? We could copy the bytes to BufferHolder directly, I think.